### PR TITLE
add smithy-ci gradle task

### PR DESCRIPTION
**Pull request by AndrewFossAWS**
### Description

> In the past, Smithy changes have been released without integration testing with Smithy-based SDK code generators. In addition, SDK code generators often fall behind in Smithy versions. As a result, we don’t have visibility on code generators’ failures until they are upgraded to the latest smithy version. 

> Upgrading versions and building all the code generators manually is also a labor-intensive task. Hence, the goal is to design a continuous integration system for testing the latest Smithy implementation with the SDK code generators.

This PR adds  `smithy-ci` gradle task to be executed by Smithy integration tests.

### CODE REVIEW

**Review by syall — APPROVED** [review]
Approving, equivalent to `make ci-test`.

File: codegen/build.gradle.kts [hunk 1]
```diff
@@ -19,3 +19,17 @@ allprojects {
         mavenCentral()
     }
 }
+
+ext {
+    this.set("smithyVersion", project.file("SMITHY_VERSION").readText().replace(System.lineSeparator(), ""))
```
Comment (isaiahvita) [inline codegen/build.gradle.kts:24]: why this change? is this some documented smithy convention to have an extra `SMITHY_VERSION` file that then gets read into `gradle.properties`?

Comment (AndrewFossAWS) [reply codegen/build.gradle.kts:24]: During the integration testing, it updates the SDK's smithy version to the latest. I could have used `sed -i "25s/.*/smithyVersion=$smithyVersion/" gradle.properties` but using line number is too fragile as dependencies get updated in `gradle.properties` Hence why the extra file seems appropriate.

Comment (isaiahvita) [reply codegen/build.gradle.kts:24]: hm im not sure im following, are you saying that with your proposed change, there will be integration testing code that updates the extra file `SMITHY_VERSION`. and if you didnt have this extra file then you would have to update `gradle.properties` via line number using `sed` because in whatever context youre in you cannot access `gradle.properties` via its normal getter/setter gradle api? 

if that understanding is correct, i dont see any code in this PR that writes to a `SMITHY_VERSION` file.
